aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,486 @@
+
+In the following example, the data model is decorated with ``DataAnnotations`` attributes, which generates validation error messages on the ``<input>`` element.  When a validation error occurs, the Validation Tag Helper displays the error message:
+
+.. literalinclude::  forms/sample/final/ViewModels/RegisterViewModel.cs
+  :language: c#
+
+..  literalinclude::  forms/sample/final/Views/Demo/RegisterValidation.cshtml
+  :language: HTML
+  :emphasize-lines: 4,6,8
+  :lines: 1-10
+
+The generated HTML:
+
+.. code-block:: HTML
+  :emphasize-lines: 2,3,8,9,12,13
+  :linenos:
+  
+  <form action="/DemoReg/Register" method="post">
+    <div class="validation-summary-valid" data-valmsg-summary="true">
+    <ul><li style="display:none"></li></ul></div>
+    Email:  <input name="Email" id="Email" type="email" value="" 
+     data-val-required="The Email field is required." 
+     data-val-email="The Email field is not a valid e-mail address." 
+     data-val="true"> <br>
+    <span class="field-validation-valid" data-valmsg-replace="true" 
+     data-valmsg-for="Email"></span><br>
+    Password: <input name="Password" id="Password" type="password" 
+     data-val-required="The Password field is required." data-val="true"><br>
+    <span class="field-validation-valid" data-valmsg-replace="true" 
+     data-valmsg-for="Password"></span><br>
+    <button type="submit">Register</button>
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+  </form>
+
+The Select Tag Helper
+-------------------------
+
+.. question One odd thing I noticed about this is that "asp-for" doesn't require the Model. prefix, but "asp-items" does. If anybody knows why this is, I'd love to hear it.  http://www.exceptionnotfound.net/tag-helpers-in-asp-net-5-an-overview/#selecttaghelper 
+
+The `Select Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/SelectTagHelper/index.html>`__ generates `select <https://www.w3.org/wiki/HTML/Elements/select>`__ and associated `option <https://www.w3.org/wiki/HTML/Elements/option>`__ elements for properties of your model. ``asp-for`` specifies the model property  name for the `select <https://www.w3.org/wiki/HTML/Elements/select>`__ element  and ``asp-items`` specifies the `option <https://www.w3.org/wiki/HTML/Elements/option>`__ elements.  For example:
+
+.. literalinclude::   forms/sample/final/Views/Home/Index.cshtml
+  :language: HTML
+  :lines: 4
+  :dedent: 3
+  
+Consider the following view model:
+
+.. literalinclude::  forms/sample/final/ViewModels/CountryViewModel.cs
+  :language: c#
+  
+The ``Index`` method initializes the ``CountryViewModel``, sets the selected country and passes it to the ``Index`` view.
+
+.. literalinclude:: forms/sample/final/Controllers/HomeController.cs
+  :language: c#
+  :lines: 8-13
+  :dedent: 6
+
+The HTTP POST ``Index`` method displays the selection:
+
+.. literalinclude:: forms/sample/final/Controllers/HomeController.cs
```

Comment:
The form submission stuff is pretty simple. Customers have the most problems with select lists and I anticipate that's where they'll need full samples. A form by itself isn't that useful. I put a note on the forms sections - see the input section for complete form submission samples.